app/src/main/java/com/android/sample/ui/authentication/SignInScreen.kt

```diff
@@ -188,7 +190,10 @@ fun SignInScreen(navigationActions: NavigationActions, userViewModel: UserViewMo
           // Show the loading animation if the user is registered
           if (Firebase.auth.currentUser != null || registered.value) {
             LoadingAnimation(
-                onFinish = { navigationActions.navigateTo(Screen.SWIPE) },
+                onFinish = {
+                  userViewModel.getCurrentUser(isConnected)
+                  navigationActions.navigateTo(Screen.SWIPE)
+                },
```

Comment:
This part hasn’t been tested yet. It might be a good idea to try returning a mocked user to satisfy the condition and properly test this part of the code. You can refer to this post for guidance on mocking the user: https://stackoverflow.com/questions/61408137/mock-implementation-of-firebase-auth-methods-after-mocking-the-firebase-module.

What do you think?

Reply:
I’ll wait to check with @CapraSnitzel since the issue is assigned to him. He might be the one handling the testing. Let me know if there’s anything else I can assist with in the meantime!